app/src/main/java/com/eveningoutpost/dexdrip/utils/LibreTrendUtil.java

```diff
@@ -70,20 +71,40 @@ public String toString() {
     }
 }
 
+class LazyLibreList extends ArrayList<LibreTrendPoint> {
```

Comment:
Why do we need this extra class? IMHO, we can simply create the array list and replace the checks like at line https://github.com/NightscoutFoundation/xDrip/blob/e65299d2f5655ef797175a5c2a54e397fa76dfb8/app/src/main/java/com/eveningoutpost/dexdrip/utils/LibreTrendUtil.java#L203 to

```java
            if (m_points.get((int)id) == null) {
                m_points.get((int)id) = new LibreTrendPoint(id, glucoseData.glucoseLevelRaw, glucoseData.flags, glucoseData.source);
            }
```

And line https://github.com/NightscoutFoundation/xDrip/blob/e65299d2f5655ef797175a5c2a54e397fa76dfb8/app/src/main/java/com/eveningoutpost/dexdrip/utils/LibreTrendUtil.java#L162 should be replaced with

```java
                if (m_points.get(i) != null) {
```

Is `m_points` used somewhere else or just dumped, if `debug_per_minute` is `true`?

And if the `m_points` is sparse, a `HashSet` can be used instead and even a `TreeSet` if ordered traversal is necessary.